### PR TITLE
Added rules to exclude the alternate files from derivative creation process. DM-170

### PR DESCRIPTION
**Pull request by lsitu**
https://lib-jira.ucsd.edu:8443/browse/DM-170

### CODE REVIEW

File: src/edu/ucsd/library/xdre/collection/CollectionHandler.java [hunk 1]
```diff
@@ -1172,7 +1172,15 @@ public static String objectPart(String symbol){
 	public static String nextBlankNodeId(){
 		return "_" + (""+random.nextInt()).replace("-", "n");
 	}
-	
+
+	public static boolean isDerivativesRequired(String fid, String use) {
+		if ( StringUtils.isBlank(fid))
+			return false;
+		return (isImage(fid, use) || isDocument(fid, use) || isVideo(fid, use) || isAudio(fid, use))
+				&& ((use == null && fid.startsWith("1."))
```
Comment (mcritchlow) [inline src/edu/ucsd/library/xdre/collection/CollectionHandler.java]: The boolean conditionals here are pretty nested. Adding to it/modifying it, might be complicated if changes are needed in the future. Could we perhaps break up the checking into two separate conditionals?

1. Is the file an image, document, video or audio file?
2. Is the file a source file or a service file being used as a source file (1.xxx)?

Comment (lsitu) [inline src/edu/ucsd/library/xdre/collection/CollectionHandler.java]: @mcritchlow Yes, it sound good and I've broken it up, see https://github.com/ucsdlib/damsmanager/pull/89/commits/1e0d3dc68c38d3b602c40276af13d068224c8b44#diff-352eb836dfc5661ee9af0408149ab1adR1176 .

**Comment by mcritchlow** [comment]
👍

**Comment by lsitu** [comment]
@mcritchlow Sorry I just realized that I switched the condition result somehow. Do you want me to revert it and create another PR?

**Comment by mcritchlow** [comment]
Yep, good catch. Let's revert and do a new PR. Thanks Longshou!

**Comment by lsitu** [comment]
The new PR is created now: https://github.com/ucsdlib/damsmanager/pull/91 . Thanks.

**Pull request merged**